src/main/java/io/github/tryexceptelse/jdex/JDex.java

```diff
@@ -1,93 +1,225 @@
+  private final File contactsFile = buildContactsFile();
 
   /**
    * Constructor for JDex main class.
    * Builds JDex object and gets it ready to be run by main loop
    */
-  public JDex(){} //skeleton
+  public JDex(String[] runtimeArgs) {
+    this.runtimeArgs = runtimeArgs;
+    rolodex = buildRolodex();
+    ui = buildUi();
+  }
 
   /**
    * Constructs a JDex main object and calls it in the main loop
+   *
    * @param args: Runtime args
    */
   public static void main(String[] args) {
-      // Skeleton test
-      JDex jDexTest = new JDex();
-      Contact testContact = new Contact();
-      Dex testDex = new Dex();
-      Gui testGui = new Gui(jDexTest, args);
-      MainCont testMainCont = new MainCont();
-      AppWindow testAppWindow = new AppWindow(); // the methods of this class
-      // cannot be skeleton tested since they inherit code from Application
-      // which expects inputs not yet available
-
-      jDexTest.setDex(testDex);
-      jDexTest.getDex();
-      jDexTest.setUi(testGui);
-      jDexTest.getUi();
-      jDexTest.buildDex();
-      jDexTest.buildUi();
-
-      testContact.setFirst("first");
-      testContact.getFirst();
-      testContact.setLast("last");
-      testContact.getLast();
-      testContact.setEmail("email string");
-      testContact.getEmail();
-      testContact.setStreetAddr("address");
-      testContact.getstreetAddr();
-      testContact.setPhoneNumber("phone");
-      testContact.getPhoneNumber();
-      testContact.setNotes("note");
-      testContact.getNotes();
-
-      testDex.getList();
-      testDex.addContact("first", "last", "email", "street address", "phone", "note");
-      testDex.save("save address");
-      testDex.load("load address");
-      testDex.search("first", "last", "email", "street address", "phone");
-
-      testGui.run(args);
-
-      testMainCont.newDex();
-      testMainCont.save();
-      testMainCont.saveAs();
-      testMainCont.load();
-
-      System.out.println("Skeleton test completed");
-
-      // Output:
+      //%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
+      // CONSTRUCTORS
+      //%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
+      //
+      // Main object:
+      //
+      JDex JDexTest = new JDex(args);
+      //
+      // Backend Objects:
+      //
+      Contact contactTest = new Contact();
+      Rolodex rolodexTest = new Rolodex();
+      //
+      // ContactEntries:
+      //
+      ContactEntry contactNotesTest = new ContactNotes();
+      ContactEntry emailAddressTest = new EmailAddress();
+      ContactEntry firstNameTest = new FirstName();
+      ContactEntry lastNameTest = new LastName();
+      ContactEntry phoneNumberTest = new PhoneNumber();
+      ContactEntry streetAddressTest = new StreetAddress();
+      //
+      // Frontend Objects:
+      //
+      AppWindow appWindowTest = new AppWindow();
+      Gui guiTest = new Gui(JDexTest);
```

Comment:
My bad, I had it named 'testJDex' before. Didn't change the case when I renamed it.